In [1]:
import math
import scipy.stats as st
import numpy as np

# Práctica 4

## 1. El propietario de Britten’s Egg Farm desea calcular la cantidad media de huevos que pone cada gallina. Una muestra de 20 gallinas indica que ponen un promedio de 20 huevos al mes, con una desviación estándar de 2 huevos al mes.

In [2]:
muestra_gallinas=20
promedio_huevos=20
desviacion=2

### a) ¿Cuál es el valor de la media de la población? ¿Cuál es el mejor estimador de este valor? 

El problema no tiene media de población, para este ejemplo usaremos una media muestral que en este caso será 20 huevos promedio

### b) Explique por qué necesita utilizar la distribución t. ¿Qué suposiciones necesita hacer?

Usamos distribución t porque nuestra muestra es pequeña (menor a 30) y porque no tenemos desviacion poblacional, solo desviacion muestral

### c) ¿Cuál es el valor de t en un intervalo de confianza de 95%?

In [3]:
grado_de_libertad=muestra_gallinas-1
confianza=0.95
alfa=0.05
valor_t=st.t.ppf(1-alfa/2,grado_de_libertad)
print(f'El valor t para el ejmplo dado es {round(valor_t,3)}')

El valor t para el ejmplo dado es 2.093


### d) Construya el intervalo de confianza de 95% de la media de población.

In [4]:
limite_superior=promedio_huevos + (valor_t*(desviacion/math.sqrt(muestra_gallinas)))
limite_inferior=promedio_huevos - (valor_t*(desviacion/math.sqrt(muestra_gallinas)))
print(f'Los intérvalos de cofianza al {confianza*100}% de la media de población son: Lim Inferior {round(limite_inferior,2)} y Lim Superior {round(limite_superior,2)}')

Los intérvalos de cofianza al 95.0% de la media de población son: Lim Inferior 19.06 y Lim Superior 20.94


### e) ¿Es razonable concluir que la media poblacional es de 21 huevos? ¿Y de 25 huevos?

No es razonable que la media poblacional sea **ni 21 ni 25 huevos**, ya que vimos que los intervalos son máximo hasta 20.94
-   Talvez podriamos redondear los 20.94 a 21 huevos, pero sería ser generoso con la respuesta.
-   Para el promedio de 25 ya esta muy lejos y no podemos considerarlo

## 2. Sean  las  siguientes  hipótesis: 𝐻0:𝜇 =100; 𝐻1: 𝜇 ≠100 ,  una  muestra aleatoria de seis elementos dio como resultado los siguientes valores: 118, 105, 112, 119, 105 y 111. ¿Puede concluir que la media poblacional es diferente de 100 con un nivel de significancia de 0.05?

In [5]:
def desviacion_estandar_muestral(datos:np.array)->float:
    n=len(datos)
    u = sum(datos) / n

    return math.sqrt(sum((x - u)**2 for x in datos) / (n - 1))

In [6]:
valores=np.array([118, 105, 112, 119, 105 , 111])
muestra=len(valores)
alpha=0.05
confianza=0.95

media_muestral=valores.sum()/muestra
desviacion_muestral=desviacion_estandar_muestral(valores)

muestra,alpha,confianza,media_muestral,desviacion_muestral

(6, 0.05, 0.95, np.float64(111.66666666666667), 6.0553007081949835)

### a) Establezca la regla de decisión.

In [11]:
grados_de_libertad=muestra-1
valor_critico_sup = st.t.ppf(1 - alpha/2, grados_de_libertad)
valor_critico_inf = st.t.ppf( alpha/2, grados_de_libertad)
print(f'Definimos que los valores críticos de decisión son {round(valor_critico_inf,4)} inferior y {round(valor_critico_sup,4)} superior')


Definimos que los valores críticos de decisión son -2.5706 inferior y 2.5706 superior


### b) Calcule el valor del estadístico de prueba.

In [13]:
mu0 = 100

error_estandar = desviacion_muestral / math.sqrt(muestra)
t_estadistico = (media_muestral - mu0) / error_estandar
print(f'El valor estadístico de prueba calculado es : {round(t_estadistico,4)}')

El valor estadístico de prueba calculado es : 4.7194


### c) ¿Cuál es su decisión en lo que se refiere a la hipótesis nula?

In [20]:
if (t_estadistico > valor_critico_sup) or (t_estadistico < valor_critico_inf):
    print("Decisión: Rechazamos la hipótesis nula H0.")
else:
    print("Decisión: No rechazamos la hipótesis nula H0.")


Decisión: Rechazamos la hipótesis nula H0.
